braille-printer-app/brf-printer.h

```diff
@@ -0,0 +1,498 @@
+
+extern int brf_JobIsCanceled(void *data);
+
+void brf_JobLog(void *data,cf_loglevel_t level,const char *message,...);
+
+typedef struct brf_spooling_conversion_s
+{
+    char *srctype;                         // Input data type
+    char *dsttype;                         // Output data type
+    cf_filter_filter_in_chain_t filters ; // List of filters with
+                                           // parameters
+} brf_spooling_conversion_t;
+
+typedef struct brf_printer_app_config_s
+{
+  // Identification of the Printer Application
+  const char *system_name;           // Name of the system
+  const char *system_package_name;   // Name of Printer Application
+                                     // package/executable
+  const char *version;               // Program version number string
+  unsigned short numeric_version[4]; // Numeric program version
+  const char *web_if_footer;         // HTML Footer for web interface
+
+  pappl_pr_autoadd_cb_t autoadd_cb;
+
+  pappl_pr_identify_cb_t identify_cb;
+
+  pappl_pr_testpage_cb_t testpage_cb;
+
+  cups_array_t *spooling_conversions;
+
+  cups_array_t *stream_formats;
+  const char *backends_ignore;
+
+  const char *backends_only;
+
+  void *testpage_data;
+
+} brf_printer_app_config_t;
+
+
+typedef struct brf_printer_app_global_data_s
+{
+  brf_printer_app_config_t *config;
+  pappl_system_t *system;
+  int num_drivers;            // Number of drivers (from the PPDs)
+  pappl_pr_driver_t *drivers; // Driver index (for menu and
+                              // auto-add)
+  char spool_dir[1024];       // Spool directory, customizable via
+                              // SPOOL_DIR environment variable
+
+} brf_printer_app_global_data_t;
+
+// Data for brf_print_filter_function()
+typedef struct brf_print_filter_function_data_s
+// look-up table
+{
+  pappl_device_t *device;                     // Device
+  const char *device_uri;                     // Printer device URI
+  pappl_job_t *job;                           // Job
+  brf_printer_app_global_data_t *global_data; // Global data
+} brf_print_filter_function_data_t;
+
+typedef struct brf_cups_device_data_s
+{
+  const char *device_uri; // Device URI
+  int inputfd,            // FD for job data input
+      backfd,             // FD for back channel
+      sidefd;             // FD for side channel
+  int backend_pid;        // PID of CUPS backend
+  double back_timeout,    // Timeout back channel (sec)
+      side_timeout;       // Timeout side channel (sec)
+
+  cf_filter_filter_in_chain_t *chain_filter; // Filter from PPD file
+  cf_filter_data_t *filter_data;             // Common data for filter functions
+  cf_filter_external_t backend_params;       // Parameters for launching
+                                             // backend via ppdFilterExternalCUPS()
+  bool internal_filter_data;                 // Is filter_data
+                                             // internal?
+} brf_cups_device_data_t;
```

Comment:
None of this is actually used (not even `filter_data`, actually), you  can remove it